### PR TITLE
check liquidity every 3mins

### PR DESCRIPTION
**Pull request by andrewdamelio**
Only sync orderBook every 3mins instead of on ZeroX events.

### CODE REVIEW

File: packages/augur-sdk/src/state/db/MarketDB.ts [hunk 1]
```diff
@@ -65,10 +67,19 @@ export class MarketDB extends DerivedDB {
...
     this.augur.events.subscribe(SubscriptionEventName.NewBlock, this.processNewBlock);
     this.augur.events.subscribe(SubscriptionEventName.TimestampSet, this.processTimestampSet);
+
+    // Don't call this interval during tests
+    if (process.env.NODE_ENV !== 'test') {
+      if (!liquidityCheckInterval) {
+        // recalc liquidity every 3mins
+        const THREE_MINS_IN_MS = 180000;
+        liquidityCheckInterval = setInterval(() => {
+          this.syncOrderBooks([]);
```
Comment (nuevoalex) [inline packages/augur-sdk/src/state/db/MarketDB.ts]: This is syncing every single market every 3 minutes. I think we should also still be listening to those events and marking which markets need to have their liquidity recalculated. Then when this gets called pass in those market ids.

Comment (andrewdamelio) [reply packages/augur-sdk/src/state/db/MarketDB.ts]: @nuevoalex ok that makes more sense, easy enough 👍

File: packages/augur-sdk/src/state/db/MarketDB.ts [hunk 2]
```diff
@@ -113,13 +132,26 @@ export class MarketDB extends DerivedDB {
...
         doc['market'] = marketId;
+        doc['liquidityDirty'] = false;
         documents.push(doc);
       }
     }
 
     await this.bulkUpsertDocuments(documents);
   }
 
+  async markMarketLiquidityAsDirty(marketId: string) {
```
Comment (nuevoalex) [inline packages/augur-sdk/src/state/db/MarketDB.ts]: What I would do instead actually is just keep this in memory and store the marketIds as a Set

File: packages/augur-sdk/src/state/db/MarketDB.ts [hunk 3]
```diff
@@ -65,10 +67,27 @@ export class MarketDB extends DerivedDB {
...
     this.augur.events.subscribe(SubscriptionEventName.NewBlock, this.processNewBlock);
     this.augur.events.subscribe(SubscriptionEventName.TimestampSet, this.processTimestampSet);
+
+    // Don't call this interval during tests
+    if (process.env.NODE_ENV !== 'test') {
+      if (!liquidityCheckInterval) {
+        // call recalc liquidity every 3mins
+        const THREE_MINS_IN_MS = 180000;
+        liquidityCheckInterval = setInterval(async () => {
+          const marketsToCheck = await this.stateDB.Markets
```
Comment (nuevoalex) [inline packages/augur-sdk/src/state/db/MarketDB.ts]: Noted below but if the market marketIds are just stored in memory as a set you can just pass those in here and destroy that set after.

File: packages/augur-sdk/src/state/db/MarketDB.ts (hunk 2 above)
Comment (andrewdamelio) [reply packages/augur-sdk/src/state/db/MarketDB.ts]: 👍  updating

**Review by nuevoalex — COMMENTED** [review]
Overall good! Think the dirty market handling can be done in a less expensive way

File: packages/augur-sdk/src/state/db/MarketDB.ts (hunk 3 above)
Comment (andrewdamelio) [reply packages/augur-sdk/src/state/db/MarketDB.ts]: 👍  updating

File: packages/augur-sdk/src/state/db/MarketDB.ts [hunk 4]
```diff
@@ -65,10 +68,24 @@ export class MarketDB extends DerivedDB {
...
-    this.augur.events.subscribe('DB:get:ZeroXOrders', (markets) => this.syncOrderBooks(markets));
+    this.augur.events.subscribe('DB:updated:ZeroXOrders', (orderEvents) => this.markMarketLiquidityAsDirty(orderEvents.market));
     this.augur.events.subscribe(SubscriptionEventName.NewBlock, this.processNewBlock);
     this.augur.events.subscribe(SubscriptionEventName.TimestampSet, this.processTimestampSet);
+
+    // Don't call this interval during tests
+    if (process.env.NODE_ENV !== 'test') {
+      if (!liquidityCheckInterval) {
+        // call recalc liquidity every 3mins
+        const THREE_MINS_IN_MS = 180000;
```
Comment (bthaile) [inline packages/augur-sdk/src/state/db/MarketDB.ts:79]: why not use block number and just estimate number of blocks as being 3 minutes. then don't need to use setInterval.

could hook into `processNewBlock` and then check liquidity every 12 blocks

Comment (nuevoalex) [reply packages/augur-sdk/src/state/db/MarketDB.ts:79]: Block time may change. the time interval is more accurate to our goal really

**Review by nuevoalex — APPROVED** [review]
👍

**Pull request merged**